src/main/java/com/atlassian/bitbucket/jenkins/internal/config/BitbucketPluginConfiguration.java

```diff
@@ -31,11 +35,21 @@ public BitbucketPluginConfiguration() {
     @Override
     public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
         req.bindJSON(this, json);
-        if (getServerList().stream().allMatch(server -> server.validate())) {
+        List<FormValidation> aggregate = getServerList()
+                .stream()
+                .map(BitbucketServerConfiguration::validate)
+                .flatMap(Collection::stream)
+                .filter(validation -> validation.kind == Kind.ERROR)
+                .collect(Collectors.toList());
+
+        if (aggregate.isEmpty()) {
             save();
             return true;
+        } else if (aggregate.size() == 1) {
+            throw new FormException(aggregate.get(0).getMessage() + FORM_INVALID_C2A, "Bitbucket Server");
```

Comment:
Can `"Bitbucket Server"` please be a constant? Also, this doesn't seem right - Isn't this meant to be the field name?